File: backend/AInterviewer/AInterviewer/DTOs/Interview.cs
using AInterviewer.Models;
using System.Security.Cryptography.X509Certificates;
using System.Text.Json.Serialization;

namespace AInterviewer.DTOs;

public record InterviewCreateRequest
(
    string CategoryName,
    string DifficultyLevel
);
public record SubmitAnswerDto
(
    Guid InterviewId,
    Guid QuestionId,
    string Answer
);
public record InterviewDto
(
    Guid Id,
    string Name,
    string CategoryName,
    int DifficultyLevel,
    DateTime CreatedAt,
    string Status,
    int Score
);
public record QuestionDto
(
    Guid Id,
    string Type,
    string Topic,
    string Content,
    int Difficulty,
    int OrderIndex
);
public class GeneratedQuestionDto
{
    [JsonPropertyName("topic")]
    public string Topic { get; set; } = "";
    [JsonPropertyName("content")]
    public string Content { get; set; } = "";
    [JsonPropertyName("difficulty")]
    public int Difficulty { get; set; }
}
public record AnswerEvaluationAndSuggestionDto(
    int Score,
    string Feedback,
    string SuggestedNextTopic
);
public record AnswerEvaluationDto(
    int Score,
    string Feedback
);

public class InterviewResultDto
{
    public int TotalScore { get; set; }
    public int CorrectAnswers { get; set; }
    public int TotalAnswers { get; set; }
    [JsonPropertyName("level")]
    public string Level { get; set; } = string.Empty;
    [JsonPropertyName("strengths")]
    public string Strengths { get; set; } = string.Empty;
    [JsonPropertyName("weaknesses")]
    public string Weaknesses { get; set; } = string.Empty;
    [JsonPropertyName("recomendations")]
    public string Recomendations { get; set; } = string.Empty;
}

File: frontend/src/types/interview.ts
export interface Interview {
  id: string;
  name: string;
  categoryName: string;
  difficultyLevel: number;
  createdAt: string;
  status: InterviewStatus;
  score: number;
  currentQuestionIndex: number;
  maxQuestions: number;
}
export const InterviewStatus = {
  InProgress: "InProgress",
  Completed: "Completed",
  NotStarted: "NotStarted"
} as const;
export type InterviewStatus = typeof InterviewStatus[keyof typeof InterviewStatus];
export interface QuestionResponse {
  id: string;
  type: string;
  topic: string;
  content: string;
  difficulty: number;
  orderIndex: number;
}

export interface SubmitAnswerRequest {
  answer: string;
}

export interface InterviewResult {
    id: string;
    interviewId: string;
    totalScore: number;
    correctAnswers: number;
    totalAnswers: number;
    level: string;
    strengths: string;
    weaknesses: string;
    recomendations: string;
}
export interface Category{
    id: string;
    name: string;
    description: string;
    interviewPrompt: string;
    evaluationPrompt: string;
    defaultDifficulty: number;
    maxQuestions: number;
    isActive: boolean;
    createdAt: Date
}